src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -778,23 +788,29 @@ private function getMergeFormData(int $orderId, int $shipmentId): array
     #[AdminSecurity("is_granted('update', 'AdminOrders')", redirectRoute: 'admin_orders_view', redirectQueryParamsToKeep: ['orderId'], message: 'You do not have permission to edit this.')]
     public function splitShipmentAction(int $orderId, int $shipmentId, Request $request): RedirectResponse
     {
-        $data = $request->get('split_shipment', []);
-        $carrier = isset($data['carrier']) ? (int) $data['carrier'] : null;
-
-        $productsRaw = $data['products'] ?? [];
-        $products = [];
+        try {
+            $data = $request->get('split_shipment', []);
+            $carrier = isset($data['carrier']) ? (int) $data['carrier'] : null;
+
+            $productsRaw = $data['products'] ?? [];
+            $products = [];
+
+            foreach ($productsRaw as $prod) {
+                if (isset($prod['selected']) && (int) $prod['selected'] === 1) {
+                    $products[] = [
+                        'id_order_detail' => (int) $prod['order_detail_id'],
+                        'quantity' => (int) $prod['selected_quantity'],
+                    ];
+                }
+            }
 
-        foreach ($productsRaw as $prod) {
-            if (isset($prod['selected']) && (int) $prod['selected'] === 1) {
-                $products[] = [
-                    'id_order_detail' => (int) $prod['order_detail_id'],
-                    'quantity' => (int) $prod['selected_quantity'],
-                ];
+            $this->dispatchQuery(new SplitShipment($shipmentId, $products, $carrier));
+        } catch (Exception $e) {
+            if ((new ReflectionClass($e))->getShortName() === 'CannotEditShipmentShippedException') {
```

Comment:
```suggestion
            if ((new ReflectionClass($e))->getName() === CannotEditShipmentShippedException::class) {
```

with the proper `use`